File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  def self.value(colors = [])
    bands = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white']
    colors.take(2).map do |color|
      bands.index(color)
    end.join.to_i
  end
end

File: javascript/resistor-color-duo/resistor-color-duo.js
export const decodedValue = (colors) => {
  const bands = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'];
  // colors.slice(0, 2);
  var a = bands.indexOf(colors[0]);
  var b = bands.indexOf(colors[1]);
  return parseInt(`${a}${b}`);
  // return Number(`${a}${b}`)
};
